src/ui/MainWindow.cpp

```diff
@@ -83,6 +83,7 @@ namespace wfl::ui
 
         m_webView.signalLoadStatus().connect(sigc::mem_fun(*this, &MainWindow::onLoadStatusChanged));
         m_webView.signalNotification().connect(sigc::mem_fun(m_trayIcon, &TrayIcon::setAttention));
+        m_webView.signalNotificationClicked().connect(sigc::mem_fun(*this, &MainWindow::onNotificationClicked));
```

Comment:
You can directly use the `onShow` function to avoid duplicate code and implement a more reliable mechanism.

Reply:
If I use `onShow`, and there are no windows, a new window is created, but isn't actually displayed -- it doesn't come to the foreground. Apparently `show` only "[flags a widget to be displayed](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Widget.html#Gtk.Widget.show)". However, if I change `show` to `present` in `onShow`, it works. Shall I do that?

```C++
    void MainWindow::onShow()
    {
        if (!is_visible())
        {
            Application::getInstance().add_window(*this);
            // This is the line I'm suggesting we change
            // show();
            present();
        }
        else
        {
            present();
        }
    }
```

Reply:
I see. Okay then, let's do it as you say. However, a bit of code improvement would be:
```cpp
if (!is_visible())
{
    Application::getInstance().add_window(*this);
}

present();
```

Reply:
Ah, yes. I should have seen that.

Alright, I'll go ahead and make the changes.